spec/requests/api/v1/participant_declarations_spec.rb

```diff
@@ -108,6 +108,43 @@ def build_params(attributes)
         end
       end
 
+      context "when declaration is rejected" do
+        let(:valid_params) do
+          {
+            participant_id: ect_profile.user.id,
+            declaration_type: "started",
+            declaration_date: (ect_profile.schedule.milestones.first.start_date + 2.days).rfc3339,
+            course_identifier: "ecf-induction",
+          }
+        end
+
+        context "when participant is withdrawn" do
+          before do
+            ect_profile.participant_profile_states.create({ state: "withdrawn", created_at: Time.zone.now - 1.hour })
+          end
+
+          it "returns 422 when participant is withdrawn" do
+            params = build_params(valid_params)
+            post "/api/v1/participant-declarations", params: params
+            expect(response.status).to eq 422
+            expect(response.body).to eq({ errors: [{ title: "Bad or missing parameters", detail: I18n.t(:declaration_on_incorrect_state) }] }.to_json)
+          end
+        end
```

Comment:
With the outer context, it reads as "when participant is withdrawn it returns 422 when participant is withdrawn".

Either remove the outside context (and put the before block inside the test), or remove `when participant is withdrawn` from `it` statement please.